include/class_loader/multi_library_class_loader.h

```diff
@@ -69,17 +69,10 @@ class MultiLibraryClassLoader
     boost::shared_ptr<Base> createInstance(const std::string& class_name)
     {
       logDebug("class_loader::MultiLibraryClassLoader: Attempting to create instance of class type %s.", class_name.c_str());
-      ClassLoaderVector active_loaders = getAllAvailableClassLoaders();
-      for(unsigned int c = 0; c < active_loaders.size(); c++)
-      {
-        ClassLoader* current = active_loaders.at(c);
-        if (!current->isLibraryLoaded())
-          current->loadLibrary();
-        if(current->isClassAvailable<Base>(class_name))
-          return(current->createInstance<Base>(class_name));
-      }
-
-      throw(class_loader::CreateClassException("MultiLibraryClassLoader: Could not create object of class type " + class_name + " as no factory exists for it. Make sure that the library exists and was explicitly loaded through MultiLibraryClassLoader::loadLibrary()"));
+      ClassLoader* loader = getClassLoaderForClass<Base>(class_name);
+      if (loader == NULL)
+        throw class_loader::CreateClassException("MultiLibraryClassLoader: Could not create object of class type " + class_name + " as no factory exists for it. Make sure that the library exists and was explicitly loaded through MultiLibraryClassLoader::loadLibrary()");
```

Comment:
That's a very good point, let's keep it as is given that it's consistent with every other error message in this package